org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/launching/JavaElementDelegate.java

```diff
@@ -224,6 +234,14 @@ protected IJavaElement getJavaElementForResource(IResource resource) {
 		}
 		return null;
 	}
+	
+	/**
+	 * Subclasses may override.
+	 * @since 2.21 
+	 */
+	protected IJavaElement getJavaElementForPackage(IPackageFragment packageFragment) {
+		return null;
```

Comment:
Why does this return null?

Reply:
I have chosen `null` as the return value to indicate that the Xbase framework does nothing by default and to emphasize that the subclasses should override that method to provide a meaningful return value.